R/tax_time_expand.R

```diff
@@ -0,0 +1,102 @@
+#' Expand taxa to interval-level pseudo-occurrences
+#'
+#' A function to generate interval-level pseudo-occurrences for taxa based on
+#' temporal ranges (e.g., the output of \code{\link{tax_range_time}}). While the
+#' resulting pseudo-occurrences should not be treated as equivalent to actual
+#' occurrence data (e.g., like that from the Paleobiology Database), such
+#' pseudo-occurrences may be useful for performing statistical analyses where
+#' the row representing a taxon must be replicated for each interval through
+#' which the taxon survived.
+#'
+#' @param taxdf \code{dataframe}. A \code{data.frame} of taxa (such as that
+#'   produced by \code{\link{tax_range_time}}) with columns for the maximum and
+#'   minimum ages (FADs and LADs). Each row should represent a unique taxon.
+#'   Additional columns may be included (e.g., taxon names, additional taxonomy,
+#'   etc) and will be included in the returned \code{data.frame}.
+#' @param max_ma \code{character}. The name of the column you wish to be treated
+#'   as the maximum limit (FADs) of the age range (e.g., "max_ma").
+#' @param min_ma \code{character}. The name of the column you wish to be treated
+#'   as the minimum limit (LADs) of the age range (e.g., "min_ma").
+#' @param scale \code{character}. Specify the desired geological timescale to
+#'   be used, either "GTS2020" or "GTS2012".
+#' @param rank \code{character}. Specify the desired stratigraphic rank. Choose
+#'   from: "stage", "epoch", "period", "era", and "eon".
+#' @param ext_orig \code{logical}. Should two additional columns be added to
+#'   identify the intervals in which taxa originated and went extinct?
+#'
+#' @return A \code{dataframe} where each row represents an interval during which
+#'   a taxon in the original user-supplied data was alive. The columns are
+#'   identical to those in the user-supplied data with additional columns
+#'   included to identify the intervals. If \code{ext_orig} is \code{TRUE},
+#'   two additional columns are added to identify in which intervals taxa
+#'   originated and went extinct.
+#' @section Developer(s):
+#'   William Gearty
+#' @section Reviewer(s):
+#'   TBD
+#' @export
+#' @examples
+#' taxdf <- data.frame(name = c("A", "B", "C"),
+#'                     max_ma = c(150, 60, 30),
+#'                     min_ma = c(110, 20, 0))
+#' tax_time_expand(taxdf)
+tax_time_expand <- function(
+    taxdf,
+    max_ma = "max_ma",
+    min_ma = "min_ma",
+    scale = "GTS2020",
+    rank = "stage",
+    ext_orig = TRUE) {
+  # Handle errors
+  if (is.data.frame(taxdf) == FALSE) {
+    stop("`taxdf` should be a dataframe")
+  }
+
+  if (!is.numeric(taxdf[, max_ma])) {
```

Comment:
Does this need to be numeric only? What if the user has interval names that can be matched to GTS2020/2012?

Reply:
Hmm...they could use look_up first, then use this function? (honestly I just didn't want to deal with that possibility)

Reply:
I think it would be better to have the function able to handle interval names directly (I believe we have done this in bin_time()). However, you're right, the user should be able to use the look_up function, and then call this function. Maybe some examples of this are sufficient.